app/jobs/data_warehouse/stale_data_check_job.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module DataWarehouse
+  class StaleDataCheckJob < BaseJob
+    REPORT_NAME = 'idp_max_ids'
+
+    def perform(timestamp)
+      data = fetch_table_max_ids_and_counts(timestamp)
+      upload_to_s3(data, timestamp)
+    end
+
+    def fetch_table_max_ids_and_counts(timestamp)
+      max_ids_and_counts(timestamp)
+    end
+
+    private
+
+    def max_ids_and_counts(timestamp)
+      active_tables = {}
+      ActiveRecord::Base.connection.tables.each do |table|
+        if table_has_id_column?(table)
+          active_tables[table] = fetch_max_id_and_count(table, timestamp)
+        end
+      end
+
+      active_tables
+    end
+
+    def table_has_id_column?(table)
+      ActiveRecord::Base.connection.columns(table).map(&:name).include?('id')
+    end
```

Comment:
can we reuse the other helper method here?
```suggestion
    def table_has_id_column?(table)
      table_has_column?(table, 'id')
    end
```